### ISSUE TITLE
Module build failed: UnhandledSchemeError when using gatsby function

### ISSUE THREAD

**Issue by Talaxy009** [gatsbyjs/gatsby#38481]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I'm trying to write a gatsby function, but it seems like webpack can't compile it.

![image](https://github.com/gatsbyjs/gatsby/assets/41264828/e21e8e03-d526-48dd-a6f0-15cea63e4fed)
![image](https://github.com/gatsbyjs/gatsby/assets/41264828/8b1b73d1-26e7-4380-9174-6c44ef7bb3f0)

I have create a repo which is base on hello-world starter and add a tiny function in it to reproduce this issue, you can find the link below.

### Reproduction Link

https://github.com/Talaxy009/gatsby-function-issue

### Steps to Reproduce

1. Clone Reproduction Link
2. Install dependencies
3. Run gatsby develop
4. Visit <http://localhost:8000/api/hello>
5. You should be able to see the error from the browser and console

### Expected Result

The `/api/hello` should return "hello"

### Actual Result

The `/api/hello` return error message from webpack

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor              
  Binaries:
    Node: 18.16.1 - ~\AppData\Local\Temp\xfs-98c1ccf6\node.CMD
    Yarn: 3.6.3 - ~\AppData\Local\Temp\xfs-98c1ccf6\yarn.CMD
    npm: 9.5.1 - D:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.10.9
  Browsers:
    Edge: Spartan (44.22621.2134.0), Chromium (116.0.1938.54)
  npmPackages:
    gatsby: ^5.12.1 => 5.12.1
```

### Config Flags

Nope

**Comment by pieh** [comment]
Confirmed. https://github.com/gatsbyjs/gatsby/pull/38489 is looking to fix this.

**Comment by pieh** [comment]
This should be fixed in `gatsby@5.12.3`

**Comment by Talaxy009** [comment]
@pieh 5.12.3 works well, thanks for your work.
